packages/common/src/utils/typeUtils.ts

```diff
@@ -56,3 +56,9 @@ export type PickRename<
 > = Omit<Type, Key> & {
   [k in RenamedKey]: Type[Key]
 }
+
+export type DeepOmit<T, K> = T extends object
```

Comment:
I think this one deserves a comment describing the exact behavior. This will remove any key matching K recursively all the way down, right?

There's some danger here of tricky mistakes where a deeply nested type uses a key that happens to match K several levels down, and it gets omitted.